### DISCUSSION TITLE
SingleNodePlugin with Dropdown and TextEditor Support

### DISCUSSION THREAD

**Discussion by girish-kadambari** [facebook/lexical#5545] [Q&A]
**Problem:**
I need a Lexical plugin (SingleNodePlugin) that enforces having only one node in the editor. Users should add text manually or select from a dropdown, replacing the existing node.

**Objective:**

Allow text input.
Provide dropdown selection.
Replace existing node on dropdown selection.
Ensure only one node in the editor.
Contribution:
Seeking contributors experienced in Lexical and React to help implement SingleNodePlugin. Contributions include code, testing, and suggestions.

**Comment by isaacfink** [comment]
@girish-kadambari Did you ever figure this out? I need the same thing, I was looking into the typeahead plugin but it does't allow me to restrict text
